modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/DiscoveryDataPacket.java

```diff
@@ -307,6 +308,27 @@ else if (binEntry.getKey() < GridComponent.DiscoveryDataExchangeType.VALUES.leng
         return res;
     }
 
+    /**
+     * @param value Value to check.
+     * @return {@code true} if value is zipped.
+     */
+    private boolean isZipped(byte[] value) {
+        return value != null && value.length > 3 && makeInt(value) == ZIP_HEADER_SIGNATURE;
+    }
+
+    /**
+     * Make int from first 4 bytes in little-endian byte order.
+     *
+     * @param b Source of bytes.
+     * @return Made int.
+     */
+    static private int makeInt(byte[] b) {
```

Comment:
Is it not possible to use BinaryPrimitives.readInt() here? Any endianness considerations?